File: leet_504_base_7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num is 0:
            return "0"
        sign = False
        s = ""
        if num < 0:
            sign = True
            num = -num
        while num:
            s += str(num % 7)
            num //= 7
        return '-' + s[::-1] if sign else s[::-1]

File: leet_504_base_7.cpp
class Solution {
public:
    string convertToBase7(int num) {

        if (!num)   return "0";

        string  s = "";//, res = "";
        long    n = num;
        bool    sign = false;

        if (num < 0)
            n = -n, sign = true;
        while (n)
        {
            s += (char) (n % 7 + '0');
            n /= 7 ;
        }
        return sign ? '-' + string(s.rbegin(), s.rend()) : string(s.rbegin(), s.rend()) ;
    }
};
